gravitee-am-ui/src/app/domain/applications/application/advanced/factors/factors.component.ts

```diff
@@ -79,40 +90,43 @@ export class ApplicationFactorsComponent implements OnInit {
 
     this.editMode = this.authService.hasPermissions(['application_settings_update']);
     this.factorService.findByDomain(this.domainId).subscribe((response) => (this.factors = [...response]));
-  }
-
-  private static getDefaultRiskAssessment() {
-    return {
-      enabled: false,
-      deviceAssessment: { enabled: false },
-      ipReputationAssessment: { enabled: false },
-      geoVelocityAssessment: { enabled: false },
-    };
+    this.challengeActive = this.application.settings.challenge.active;
   }
 
   patch(): void {
+    console.log('risk ', this.riskAssessment);
     const data = {
       factors: this.application.factors,
       settings: {
         riskAssessment: this.riskAssessment,
         mfa: {
           stepUpAuthenticationRule: this.mfaStepUpRule,
+          // stepUpAuthentication: {
+          //   active: true,
+          //   stepUpAuthenticationRule: this.mfaStepUpRule,
+          // },
```

Comment:
**question**: I am confused here, which stepUp attribute is the one to use starting from 4.3. stepUpAuthenticationRule or the stepUpAuthentication?

Reply:
Outdated